src/data/pokemon/base_stats.h

```diff
@@ -3536,7 +3809,9 @@ const struct BaseStats gBaseStats[] =
         .eggGroup1 = EGG_GROUP_FIELD,
         .eggGroup2 = EGG_GROUP_FIELD,
         .abilities = {ABILITY_WATER_ABSORB, ABILITY_WATER_ABSORB},
```

Comment:
Yes, Egg did this because an Eevee with the second ability would evolve to have `ABILITY_NONE`. The best course imo should be fixing the logic for abilities upon evolving